python/hail/expr/functions.py

```diff
@@ -945,34 +911,96 @@ def parse_variant(s, reference_genome=None):
                           TVariant(reference_genome), s._indices, s._aggregations, s._joins, s._refs)
 
 
-@typecheck(i=expr_int32)
-def call(i):
-    """Construct a call expression from an integer or integer expression.
+@typecheck(phased=expr_bool,
+           alleles=expr_int32)
+def call(phased, *alleles): # FIXME: Python 3 allows optional kwarg after varargs
+    """Construct a call expression.
+
+    Examples
+    --------
+    .. doctest::
+
+        >>> eval_expr(functions.call(True, 1, 0))
+        Call(alleles=[1, 0], phased=True)
+
+    Parameters
+    ----------
+    phased : :obj:`bool`
+        If ``True``, preserve the order of `alleles`.
+    alleles : variable-length args of :obj:`int` or :class:`.Int32Expression`
+        List of allele indices.
+
+    Returns
+    -------
+    :class:`.CallExpression`
+    """
+
+    indices, aggregations, joins, refs = unify_all(phased, *alleles)
+    return construct_expr(ApplyMethod('Call', phased._ast, *[a._ast for a in alleles]), TCall(), indices, aggregations, joins, refs)
```

Comment:
You should check the number of arguments here, otherwise I think you will get a function registry match error.